lex.go

```diff
@@ -244,8 +246,10 @@ func lexVariable(l *lexer) stateFn {
 			// strip arch from source_arch like constructs
 			witharch := strings.SplitN(variable, "_", 2)
 			if len(witharch) == 2 {
-				if _, ok := archs[witharch[1]]; ok {
-					variable = witharch[0]
+				for _, arch := range l.pkgbuild.Arch {
```

Comment:
I'm not a fan of this circular dependency where the lexer depends on a `*PKGBUILD`. Can't we just drop this check from the lexer and then check if we get back a valid arch in the parser?